compiler/env/TypedAllocator.hpp

```diff
@@ -63,6 +63,7 @@ class typed_allocator<void, Alloc>
       return !(left == right);
       }
 
+   untyped_allocator backingAllocator() { return _backingAllocator; }
```

Comment:
This sits strangely to me.  I wonder if this shouldn't instead be implemented as a conversion operator.